### PR TITLE
Feature/docker2

### PR DESCRIPTION
**Pull request by keiranmraine**
Replaces #9 as significant changes to target of merge.

I've created a docker file which automatically add the fonts and exposes them via an env variable.

Running this shows that the test case doesn't exit with a 0 exit code (standard for success) when using `runCompareTwoGenomesColinear.sh`.

```bash
docker build .
cd $SCRATCHAREA
docker run -ti --rm -v $PWD:/data $(docker images -q | head -n 1) bash
# in container
cd /data && cp /opt/xmatchview/tests/test/* .
runCompareTwoGenomesColinear.sh FTL1_pa.fa FTL1_ss.fa 200 90 10 2 2 FTL1_pa.gff FTL1_ss.gff minimap2 .
...
2374 out of 2375
done.
Drawing repeats...
JN039333.1_Picea_abies (50-1213) hits KT263970.1_Picea_sitchensis  ::  mismatch 27.66 target(90)  block 1181 target (10) 
JN039333.1_Picea_abies (386-650) hits KT263970.1_Picea_sitchensis  ::  mismatch 60.79 target(90)  block 291 target (10) 
Saving xmv-FTL1_pa.fa_vs_FTL1_ss.fa.paf_m90_b10_r2_c2.png...
done.
```

Resulting image is complete and matches example:

xmv-FTL1_pa fa_vs_FTL1_ss fa paf_m90_b10_r2_c2

I also fixed the non-zero exit code from the python script (sorry my editor removed trailing whitespace, edit the MR URL appending `?w=1` to hide).

### CODE REVIEW

**Comment by keiranmraine** [comment]
If you are interested I'd be happy to help you get this setup to auto-build the docker image into quay.io.  Dockerhub isn't the best place anymore as they will purge images that aren't from paid accounts if they are not used frequently.

File: runCompareTwoGenomesColinear.sh [hunk 1]
```diff
@@ -16,23 +16,39 @@ if [ $# -ne 11 ]; then
         exit 1
 fi
 
-echo Running: $(basename $0) $1 $2 $3 $4 $5 $6 $7 $8 $9 ${10} ${11}
+if [ ${11} == '.' ]; then
```
Comment (lcoombe) [inline runCompareTwoGenomesColinear.sh:19]: What do you think about adding a comment to the 'help' printout about using `.` when the user doesn't want to specify the optional arguments? Just thinking that might not be obvious to all.

File: runCompareTwoGenomesColinear.sh [hunk 2]
```diff
@@ -16,23 +16,39 @@ if [ $# -ne 11 ]; then
...
+echo Running: $(basename $0) $1 $2 $3 $4 $5 $6 $7 $8 $9 ${10} ${XM_FONTS}
 
-echo "Make sure xmatchview.py, cross_match and minimap2 are in your PATH"
+FEATURE_OPTS=""
+if [ $8 != '.' ]; then
+   FEATURE_OPTS=" -y $8"
+fi
+if [ $9 != '.' ]; then
+   FEATURE_OPTS=" -y $9"
+fi
```
Comment (lcoombe) [inline runCompareTwoGenomesColinear.sh:35]: I believe that to maintain the same behaviour as before, `$9` should be ` -e $9`. Also, looks like you are overwriting the `FEATURE_OPTS` variable if both `$8` and `$9` are specified?

**Review by lcoombe — COMMENTED** [review]
Thanks for this work, @keiranmraine! And sure, if you could help us out with auto-building this image to quay.io that would be great. Bioconda already auto-builds a xmatchview image there, but it wouldn't include the font support that you are including here.

File: runCompareTwoGenomesColinear.sh (hunk 1 above)
Comment (keiranmraine) [reply runCompareTwoGenomesColinear.sh:19]: I've added some comments to the help section

File: runCompareTwoGenomesColinear.sh (hunk 2 above)
Comment (keiranmraine) [reply runCompareTwoGenomesColinear.sh]: Yes, that's a silly mistake, fixed in recent push.

**Comment by keiranmraine** [comment]
Dockstore has a good [guide on how to link your github repo to a quay.io build](https://docs.dockstore.org/en/develop/getting-started/getting-started-with-cwl.html#building-on-quay-io).  The main thing to ensure is that you only set the build trigger to act on new tags only (regex `tags/.+`).  You can manually trigger a build against any branch if needed.

The new image is ~30MB (25%) smaller than the bioconda one.

Dockstore.org is a tool discovery system that creates hard links between code, tool/workflow descriptions and the images.

**Review by lcoombe — APPROVED** [review]
Thanks for this, @keiranmraine  - Look good to me now! And thanks for the info about the docker image builds for quay.io -- I'll take a look at that.

Any comments, @warrenlr , or are we good to merge?

**Pull request merged**